### PR TITLE
Added custom field support

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lib/auto-login.js [hunk 1]
```diff
@@ -178,6 +168,41 @@ module.exports = function setupAutoLogin(dependencies, params, callback) {
...
+      	done(null, uid, data);
+      });
+
+  	}, (uid, data, done) => {
+
+  		if(data && data.musicoinprofile) {
+  			return done(null, uid);
+  		}
+
+  		let customFields = {musicoinprofile: `${appURL}/artist/${user.profileAddress}`};
```
Comment (hayrullahcansu) [inline lib/auto-login.js:189]: does profile address just provide for artist?  or this is valid for all users?

File: library.js [hunk 2]
```diff
@@ -21,6 +21,14 @@ exports.overrideLoginController = function overrideLoginController(params, callb
 
 };
 
+exports.updateProfileHook = function updateProfileHook(params, callback) {
```
Comment (hayrullahcansu) [inline library.js]: is this just for logging?

File: lib/auto-login.js [hunk 3]
```diff
@@ -178,6 +168,41 @@ module.exports = function setupAutoLogin(dependencies, params, callback) {
...
+  		db.setObject(createObjectKey(uid, namespace), customFields, (error) => done(error, uid));
```
Comment (hayrullahcansu) [inline lib/auto-login.js:190]: that's good way!

**Review by hayrullahcansu — APPROVED** [review]
good work. i can't see conflicts and anyproblem

File: lib/auto-login.js (hunk 1 above)
Comment (mudivili) [reply lib/auto-login.js:189]: For all users, I believe. This is better answered by @isaac or @Varunram

Comment (hayrullahcansu) [reply lib/auto-login.js:189]: api can check a user is artist or not? if we want to valid for just artists. we can check like this
`if(user.isArtist) {  
     let customFields = {musicoinprofile: '${appURL}/artist/${user.profileAddress}'};
}else callback();
`

Comment (hayrullahcansu) [reply lib/auto-login.js:189]: @immartian , @Varunram @livey0u we need to decide about this.

Comment (immartian) [reply lib/auto-login.js:189]: we can support all users, no need to restrict to artists only(or verified artist)

Comment (hayrullahcansu) [reply lib/auto-login.js:189]: okay, we can merge.

**Pull request merged**